internal/reports/csv.go

```diff
@@ -0,0 +1,437 @@
+package reports
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/kubev2v/migration-planner/internal/store/model"
+	"github.com/kubev2v/migration-planner/api/v1alpha1"
+)
+
+// GenerateCSVContent generates CSV content from source.Inventory data
+func GenerateCSVContent(source *model.Source, reportType string, includeWarnings bool) string {
+	// Check if inventory data exists and get the underlying value
+	if source.Inventory == nil {
+		return generateEmptyReport(source)
+	}
+
+	// Get the actual inventory data from the JSONField
+	inventoryData, err := source.Inventory.Value()
+	if err != nil {
+		return generateEmptyReport(source)
+	}
+	
+	// The Value() method returns []byte (JSON data), so we need to unmarshal it
+	var inv v1alpha1.Inventory
+	if inventoryBytes, ok := inventoryData.([]byte); ok {
```

Comment:
there's no need to marshal it and unmarshal it again. 
please see https://github.com/kubev2v/migration-planner/blob/ce1fc6c7105e53b68a393c7459cab41ebe31167a/internal/store/model/inventory_stats.go#L58